registry/registry.go

```diff
@@ -29,6 +30,9 @@ import (
 // used in a Deployment Manager configuration. There can be multiple
 // registry implementations.
 type Registry interface {
+	// Also handles http.Client.Do method for authenticated File accesses
+	util.HTTPDoer
```

Comment:
I was not quite sure how the client would be managing these packages, etc. so thought that the most generic way would be just fine, since we only expose them at the higher level as GET. If you feel strongly about this, I can change it to GET methods.